### ISSUE TITLE
Alerting does not work if record definitions are used in Alert rule

### ISSUE THREAD

**Issue by DeathBorn** [kevinjqiu/pat#8]
the tool does not trigger the alert if record definition is used in Alert rule. It shows empty result.

**Comment by kevinjqiu** [comment]
@DeathBorn Could you provide a sample test file that illustrate the problem?

**Comment by DeathBorn** [comment]
rules.yamls
```yaml
groups:
  - name: prometheus.rules
    rules:
      - record: httprequestratelow_record
        expr: http_requests{group="canary", job="app-server"} < 100 
      - alert: HTTPRequestRateLow
        expr: httprequestratelow_record
        for: 1m
        labels:
          severity: critical
```

test.yml
```yaml
name: Test HTTP Requests too low alert
rules:
  fromFile: rules.yaml
fixtures:
  - duration: 5m
    metrics:
      - http_requests{job="app-server", instance="0", group="canary", severity="overwrite-me"}	75 85  95 105 105  95  85
      - http_requests{job="app-server", instance="1", group="canary", severity="overwrite-me"}	80 90 100 110 120 130 140
assertions:
  - at: 0m
    expected:
      - alertname: HTTPRequestRateLow
        alertstate: pending
        group: canary
        instance: "0"
        job: app-server
        severity: critical
      - alertname: HTTPRequestRateLow
        alertstate: pending
        group: canary
        instance: "1"
        job: app-server
        severity: critical
    comment: |-
      At 0m, the alerts met the threshold but has not met the duration requirement. Expect the alert to be pending
```
pat test.yml
```
=== RUN   Test_HTTP_Requests_too_low_alert_at_0m
--- FAIL: Test_HTTP_Requests_too_low_alert_at_0m (0.00s)
	ruletest.go:91:
			Error Trace:	ruletest.go:91
			Error:      	Not equal:
			            	expected: 2
			            	actual  : 0
			Test:       	Test_HTTP_Requests_too_low_alert_at_0m
			Messages:   	Alert count does not match expected
	ruletest.go:212:
			Error Trace:	ruletest.go:212
			            				ruletest.go:92
			Error:      	Not equal:
			            	expected: []map[string]string{map[string]string{"group":"canary", "instance":"0", "job":"app-server", "severity":"critical", "__name__":"ALERTS", "alertname":"HTTPRequestRateLow", "alertstate":"pending"}, map[string]string{"alertstate":"pending", "group":"canary", "instance":"1", "job":"app-server", "severity":"critical", "__name__":"ALERTS", "alertname":"HTTPRequestRateLow"}}
			            	actual  : []map[string]string(nil)

			            	Diff:
			            	--- Expected
			            	+++ Actual
			            	@@ -1,21 +1,2 @@
			            	-([]map[string]string) (len=2) {
			            	- (map[string]string) (len=7) {
			            	-  (string) (len=8) "__name__": (string) (len=6) "ALERTS",
			            	-  (string) (len=9) "alertname": (string) (len=18) "HTTPRequestRateLow",
			            	-  (string) (len=10) "alertstate": (string) (len=7) "pending",
			            	-  (string) (len=5) "group": (string) (len=6) "canary",
			            	-  (string) (len=8) "instance": (string) (len=1) "0",
			            	-  (string) (len=3) "job": (string) (len=10) "app-server",
			            	-  (string) (len=8) "severity": (string) (len=8) "critical"
			            	- },
			            	- (map[string]string) (len=7) {
			            	-  (string) (len=8) "__name__": (string) (len=6) "ALERTS",
			            	-  (string) (len=9) "alertname": (string) (len=18) "HTTPRequestRateLow",
			            	-  (string) (len=10) "alertstate": (string) (len=7) "pending",
			            	-  (string) (len=5) "group": (string) (len=6) "canary",
			            	-  (string) (len=8) "instance": (string) (len=1) "1",
			            	-  (string) (len=3) "job": (string) (len=10) "app-server",
			            	-  (string) (len=8) "severity": (string) (len=8) "critical"
			            	- }
			            	-}
			            	+([]map[string]string) <nil>

			Test:       	Test_HTTP_Requests_too_low_alert_at_0m
FAIL
```

**Comment by kevinjqiu** [comment]
Fixed by #10 
```
$ cat test.yaml 
name: Test HTTP Requests too low alert
rules:
  fromLiteral: |-
    groups:
      - name: prometheus.rules
        rules:
          - record: httprequestratelow_record
            expr: http_requests{group="canary", job="app-server"} < 100 
            labels:
              severity: critical
          - alert: HTTPRequestRateLow
            expr: httprequestratelow_record
            for: 1m
fixtures:
  - duration: 5m
    metrics:
      - http_requests{job="app-server", instance="0", group="canary", severity="overwrite-me"}	75 85  95 105 105  95  85
      - http_requests{job="app-server", instance="1", group="canary", severity="overwrite-me"}	80 90 100 110 120 130 140
assertions:
  - at: 0m
    expected:
      - __name__: httprequestratelow_record
        group: canary
        instance: "0"
        job: app-server
        severity: critical
      - __name__: httprequestratelow_record
        group: canary
        instance: "1"
        job: app-server
        severity: critical
    comment: |-
      At 0m, the alerts met the threshold but has not met the duration requirement. Expect the alert to be pending

$ ./pat test.yaml 
=== RUN   Test_HTTP_Requests_too_low_alert_at_0m
--- PASS: Test_HTTP_Requests_too_low_alert_at_0m (0.00s)
PASS
```

**Comment by alexey-medvedchikov** [comment]
Thank you, it works for us now. 👍
